### PR TITLE
Solution toggle aria-label missing in initial load state

### PR DESCRIPTION
**Pull request by PiotrKozlowski**
Add `aria-label="show solution"` in initial load state
Transoform content before rendering

### CODE REVIEW

File: src/app/content/components/Page/PageComponent.tsx [hunk 1]
```diff
@@ -29,9 +31,14 @@ export default class PageComponent extends Component<PagePropTypes> {
...
 
-  public getCleanContent = () => {
+  public getTrasnformedContent = () => {
     const {book, page, services} = this.props;
-    return getCleanContent(book, page, services.archiveLoader, contentLinks.reduceReferences(this.props.contentLinks));
+    const cleanContent = parser.parseFromString(
+      getCleanContent(book, page, services.archiveLoader, contentLinks.reduceReferences(this.props.contentLinks)),
+      'text/html'
+    );
+    transformContent(cleanContent.body, this.props.intl);
```
Comment (TomWoodward) [inline src/app/content/components/Page/PageComponent.tsx]: the logic inside `transformContent` is depending on the document global and is failing in prerendering, we need to pass in the document that `parseFromString` returns

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/content/components/Page/contentDOMTransformations.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import { HTMLButtonElement, HTMLElement } from '@openstax/types/lib.dom';
+import { Document, HTMLButtonElement, HTMLElement } from '@openstax/types/lib.dom';
 import { IntlShape } from 'react-intl';
 import { assertDefined, assertNotNull } from '../../../utils';
 
 // from https://github.com/openstax/webview/blob/f95b1d0696a70f0b61d83a85c173102e248354cd
 // .../src/scripts/modules/media/body/body.coffee#L123
-export const transformContent = (rootEl: HTMLElement, intl: IntlShape) => {
+export const transformContent = (document: Document, intl: IntlShape) => {
+  const rootEl = document.body;
```
Comment (TomWoodward) [inline src/app/content/components/Page/contentDOMTransformations.ts]: can we pass element separately? i feel like this is a big assumption that the document is only going to contain things we want to transform

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Comment by PiotrKozlowski** [comment]
@TomWoodward I think now this is correct, but there were problems with deploying so I'm not totally sure if the output from Travis is valid.

File: src/app/content/components/Page/contentDOMTransformations.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
-import { HTMLButtonElement, HTMLElement } from '@openstax/types/lib.dom';
+import { Document, HTMLButtonElement, HTMLElement } from '@openstax/types/lib.dom';
 import { IntlShape } from 'react-intl';
 import { assertDefined, assertNotNull } from '../../../utils';
 
 // from https://github.com/openstax/webview/blob/f95b1d0696a70f0b61d83a85c173102e248354cd
 // .../src/scripts/modules/media/body/body.coffee#L123
-export const transformContent = (rootEl: HTMLElement, intl: IntlShape) => {
+// We are passing Document because it is required to prerender.
+export const transformContent = (_document: Document, rootEl: HTMLElement, intl: IntlShape) => {
```
Comment (TomWoodward) [inline src/app/content/components/Page/contentDOMTransformations.ts]: you're passing document but you're not using it for anything, you need to pass the reference down so it is used in places like this https://github.com/openstax/rex-web/pull/449/files#diff-3684e43af97114ebe9e858e13d39add5R56

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/app/content/components/Page/contentDOMTransformations.ts (hunk 3 above)
Comment (PiotrKozlowski) [reply src/app/content/components/Page/contentDOMTransformations.ts]: Omg, now this makes sense. I could sware that `document` was not used in this file (I had to mistype it when I was searching) and I thought that prerendering script is somehow getting access to this...

File: src/app/content/components/Page/contentDOMTransformations.ts [hunk 4]
```diff
@@ -44,7 +45,7 @@ function addScopeToTables(rootEl: HTMLElement) {
 }
 
 // Wrap title and content elements in header and section elements, respectively
-function wrapElements(rootEl: HTMLElement) {
+function wrapElements(document: Document, rootEl: HTMLElement) {
```
Comment (TomWoodward) [inline src/app/content/components/Page/contentDOMTransformations.ts]: now that you're passing this in the `assertDefined` is no longer necessary

**Review by TomWoodward — CHANGES_REQUESTED** [review]

**Review by TomWoodward — APPROVED** [review]

**Pull request merged**